main.tf

```diff
@@ -13,8 +13,13 @@ locals {
   private_key_filename = "${var.ssh_public_key_path}/${module.label.id}${var.private_key_extension}"
 }
 
+resource "null_resource" "import or generate key" {
+  count                                                                     = "${var.generate_ssh_key == "true" && var.import_ssh_key == "true" ? 1 : 0}"
+  "You cannot generate and import a key, you can only do one or the other." = true
```

Comment:
??

Reply:
I believe this is a pattern he invented to deliberately cause terraform to barf with an error. @Jamie-BitFlight can you share an example of what the output looks like in practice (albeit this is a bit unconventional for us)

Reply:
pernaps we should create a `terraform-null-validator` module that takes as input a condition that should be true, and if it's false, it it results in the above error condition.

Reply:
this is a good idea to have some kind of `terraform-null-validator`.
but the assertion `You cannot generate and import a key, you can only do one or the other` is not correct since the module can already generate and import a key.
Please see the comment below.

Reply:
Hi guys,
Erik is correct, its a assertion test I invented for a problem i had a few versions of terraform back, and now its used around the community.
It works in Terraform because of the way the 'plan' validator works.

During the plan phase, the 'count' is calculated.
And only if it is a positive number will the other values inside the resource get validated.
The error thrown, is a syntax error. But in Syntax errors, it displays the name of the 'key' in the resource, which means you can be descriptive.
```
Error: module.ssh_key_pair.null_resource.import or generate key: : invalid or unknown key: You cannot generate and import a key, you can only do one or the other.

```

I quickly made a module that looks like this to test it:
```hcl
variable "failure_text" {
  default = "An the values didn't have the values needed"
}
variable "conditional" { default = true }

resource "null_resource" "ASSERTION_TEST_FAILED" {
  count = "${var.conditional ? 1 : 0}"
  "${var.failure_text}" = true
}
```

And ran it by doing this
```hcl
module "assert" {
  source = "../terraform-null-assert"
}
```

and got this output from `terraform plan`
```
Error: module.assert.null_resource.ASSERTION_TEST_FAILED: : invalid or unknown key: An the values didn't have the values needed

```